MekHQ/src/mekhq/campaign/Campaign.java

```diff
@@ -5761,26 +5762,30 @@ public int totalBonusParts() {
 
     public int findAtBPartsAvailabilityLevel(IAcquisitionWork acquisition, StringBuilder reportBuilder) {
         AtBContract contract = (acquisition != null) ? getAttachedAtBContract(acquisition.getUnit()) : null;
+        
         /*
          * If the unit is not assigned to a contract, use the least restrictive active
          * contract. Don't restrict parts availability by contract if it has not started.
          */
-        for (Mission m : getMissions()) {
-            if (m.isActive() && m instanceof AtBContract && ((AtBContract) m).getStartDate().isBefore(currentDay)) {
-                if (null == contract
-                        || ((AtBContract) m).getPartsAvailabilityLevel() > contract.getPartsAvailabilityLevel()) {
-                    contract = (AtBContract) m;
+        if (hasActiveContract() && (contract == null)) {
```

Comment:
contact == null should be impossible for hasActiveContact, otherwise it is a bug in hasActiveContact.